jsonschema/json.go

```diff
@@ -27,7 +27,7 @@ type Definition struct {
 	// one element, where each element is unique. You will probably only use this with strings.
 	Enum []string `json:"enum,omitempty"`
 	// Properties describes the properties of an object, if the schema type is Object.
-	Properties map[string]Definition `json:"properties,omitempty"`
+	Properties map[string]Definition `json:"properties"`
```

Comment:
@royalrick Thanks for kind PR! Even if you remove 'omitempty', an error will be returned due to OpenAI API's specification where Properties are required.

Reply:
I tried remove 'omitempty', and it works.

Reply:
remove 'omitempty' and init properties like `make(map[string]any), it will works.

Reply:
Thanks! It seems like initializing properties as an empty object and sending it results in success.

**failed:**
```
$ curl https://api.openai.com/v1/chat/completions -u :$OPENAI_API_KEY -H 'Content-Type: application/json' -d '{
                "model": "gpt-3.5-turbo-0613",
                "messages": [
                  {"role": "user", "content": "What is the weather like in Boston?"}
                ],
                "functions": [
                  {
                    "name": "get_current_weather",
                    "parameters": {
                      "type": "object"
                    }
                  }
                ]
              }'
{
  "error": {
    "message": [
      "Invalid schema for function 'get_current_weather': In context=(), object schema missing properties"
    ],
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}
```

**success:**

```
$ curl https://api.openai.com/v1/chat/completions -u :$OPENAI_API_KEY -H 'Content-Type: application/json' -d '{
                "model": "gpt-3.5-turbo-0613",
                "messages": [
                  {"role": "user", "content": "What is the weather like in Boston?"}
                ],
                "functions": [
                  {
                    "name": "get_current_weather",
                    "parameters": {
                      "type": "object",
                      "properties": {}
                    }
                  }
                ]
              }'
{
  "id": "chatcmpl-***************************************",
  "object": "chat.completion",
  "created": 1688129746,
  "model": "gpt-3.5-turbo-0613",
  "choices": [
    {
      "index": 0,
      "message": {
        "role": "assistant",
        "content": null,
        "function_call": {
          "name": "get_current_weather",
          "arguments": "{}"
        }
      },
      "finish_reason": "function_call"
    }
  ],
  "usage": {
    "prompt_tokens": 40,
    "completion_tokens": 8,
    "total_tokens": 48
  }
}
```